src/linter/block.go

```diff
@@ -1558,6 +1558,26 @@ func (b *BlockWalker) handleForeach(s *stmt.Foreach) bool {
 		b.propagateFlags(ctx)
 	}
 
+	key, ok := s.Key.(*node.SimpleVar)
+	if !ok {
+		return false
+	}
+	if IsDiscardVar(key.Name) {
+		return false
+	}
+
+	_, ok = b.unusedVars[key.Name]
+	if ok {
+		variable, ok := s.Variable.(*node.SimpleVar)
+		if !ok {
+			return false
+		}
+
+		delete(b.unusedVars, key.Name)
+
+		b.r.Report(s.Key, LevelError, "foreachSimplify", "foreach key $%s is unused, can simplify $%s => $%s to just $%s", key.Name, key.Name, variable.Name, variable.Name)
```

Comment:
It looks like it shouldn't be a new diagnostic as it checks the same thing as before. The only difference is that it tries to give more human-readable (and more actionable) warning message.

@YuriyNasretdinov what do you think about reporting this as "unused"?

Reply:
Yeah, I guess unused would be appropriate here.